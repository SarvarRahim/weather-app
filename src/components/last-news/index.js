export * from './last-news.component';
