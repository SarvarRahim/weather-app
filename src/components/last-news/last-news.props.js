/* <LastNews /> */

import { dateFormatByMonth } from '@utils/date-format';
import { useEffect, useMemo } from 'react';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { commonActions } from 'src/store/common/common.slice';

export const useNewsLastProps = () => {
  const dispatch = useDispatch();
  const latestNews = useSelector(state => state.common.latestNews);

  const day = useMemo(() => {
    if (!latestNews) return '';
    const date = new Date(latestNews.created_at);
    return dateFormatByMonth(date);
  }, [latestNews]);

  useEffect(() => {
    dispatch(commonActions.fetchLatestNews());
  }, [dispatch]);

  return {
    latestNews,
    day
  };
};
