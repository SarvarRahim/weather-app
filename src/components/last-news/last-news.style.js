import styled from 'styled-components';

export const LastNewsContainer = styled.a`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  min-height: 215px;
  padding: 24px;
  box-shadow: 0px 4px 15px rgba(0, 0, 0, 0.05);
  border-radius: 4px;
  background-color: #161009;
  text-decoration: none;
  background-image: linear-gradient(
      0deg,
      rgba(0, 0, 0, 0.7),
      rgba(0, 0, 0, 0.7)
    ),
    url(${props => props.lastNewsBanner});
`;

export const LastNewsSpace = styled.div`
  background-color: #ffffff;
`;

export const LastNewsTitle = styled.h2`
  font-weight: 600;
  font-size: 20px;
  line-height: 27px;
  color: #fff;
`;

export const LastNewsAuthorWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-top: 12px;
`;

export const LastNewsAuthorImg = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 50%;
  object-fit: cover;
`;

export const LastNewsBottomWrapper = styled.div`
  margin-left: 16px;
`;

export const LastNewAuthorName = styled.h3`
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  color: #ffffff;
`;

export const LastNewsDate = styled.time`
  font-size: 14px;
  line-height: 19px;
  color: rgba(255, 255, 255, 0.5);
`;
