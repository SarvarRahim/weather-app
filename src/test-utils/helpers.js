import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
