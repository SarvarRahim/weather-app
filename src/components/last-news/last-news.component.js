import { useNewsLastProps } from './last-news.props';
import {
  LastNewAuthorName,
  LastNewsAuthorImg,
  LastNewsAuthorWrapper,
  LastNewsBottomWrapper,
  LastNewsContainer,
  LastNewsDate,
  LastNewsSpace,
  LastNewsTitle
} from './last-news.style';
import { hoc } from '@utils/hoc';
import { config } from 'src/config';

export const LastNews = hoc(
  useNewsLastProps,
  ({ latestNews, day, ...rest }) => {
    return (
      <LastNewsSpace {...rest}>
        <LastNewsContainer
          lastNewsBanner={latestNews?.banner?.url}
          href={`${config.landingUrl}/blog/${latestNews?.slug}`}
          target='_blank'
        >
          <LastNewsTitle data-testid='last-news-title'>
            {latestNews?.title?.uz}
          </LastNewsTitle>
          <LastNewsAuthorWrapper>
            <LastNewsAuthorImg
              src={latestNews?.thumbnail?.url}
              alt={latestNews?.title?.uz}
              width={40}
              height={40}
            />
            <LastNewsBottomWrapper>
              <LastNewAuthorName data-testid='last-news-author'>
                {latestNews?.author?.uz}
              </LastNewAuthorName>
              <LastNewsDate data-testid='last-news-day'>{day}</LastNewsDate>
            </LastNewsBottomWrapper>
          </LastNewsAuthorWrapper>
        </LastNewsContainer>
      </LastNewsSpace>
    );
  }
);
