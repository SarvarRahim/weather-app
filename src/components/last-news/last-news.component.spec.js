import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';
import { LastNews } from './last-news.component';

const news = {
  title: {
    uz: 'Ohirgi yangilik'
  },
  banner: {
    url: 'https://via.placeholder.com/300x150'
  },
  thumbnail: {
    url: 'https://via.placeholder.com/40x40'
  },
  slug: 'ohirgi-yangilik',
  author: {
    uz: "O'zbekiston Respublikasi"
  }
};

const day = '12/12/2020';

const LAST_NEWS_PROPS = {
  latestNews: news,
  day,
  'data-testid': 'last-news-component'
};

describe('Common Context', () => {
  test('Should render LastNews component', async () => {
    const { asFragment } = customRender(<LastNews {...LAST_NEWS_PROPS} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('last-news-component')).toBeInTheDocument();
    expect(screen.getByTestId('last-news-title')).toHaveTextContent(
      news.title.uz
    );
    expect(screen.getByTestId('last-news-author')).toHaveTextContent(
      news.author.uz
    );
    expect(screen.getByTestId('last-news-day')).toHaveTextContent(day);
  });
});
