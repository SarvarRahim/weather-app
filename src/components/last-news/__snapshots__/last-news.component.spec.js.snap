// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Context Should render LastNews component 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 215px;
  padding: 24px;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.05);
  border-radius: 4px;
  background-color: #161009;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-image: linear-gradient( 0deg, rgba(0,0,0,0.7), rgba(0,0,0,0.7) ), url(https://via.placeholder.com/300x150);
}

.c0 {
  background-color: #ffffff;
}

.c2 {
  font-weight: 600;
  font-size: 20px;
  line-height: 27px;
  color: #fff;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 12px;
}

.c4 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  object-fit: cover;
}

.c5 {
  margin-left: 16px;
}

.c6 {
  font-weight: 600;
  font-size: 16px;
  line-height: 22px;
  color: #ffffff;
}

.c7 {
  font-size: 14px;
  line-height: 19px;
  color: rgba(255,255,255,0.5);
}

<div
    class="c0"
    data-testid="last-news-component"
  >
    <a
      class="c1"
      href="https://landing.najottalim.uz/blog/ohirgi-yangilik"
      target="_blank"
    >
      <h2
        class="c2"
        data-testid="last-news-title"
      >
        Ohirgi yangilik
      </h2>
      <div
        class="c3"
      >
        <img
          alt="Ohirgi yangilik"
          class="c4"
          height="40"
          src="https://via.placeholder.com/40x40"
          width="40"
        />
        <div
          class="c5"
        >
          <h3
            class="c6"
            data-testid="last-news-author"
          >
            O'zbekiston Respublikasi
          </h3>
          <time
            class="c7"
            data-testid="last-news-day"
          >
            12/12/2020
          </time>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
